static/js/index.js

```diff
@@ -157,90 +142,73 @@ async function initializeIndexAutocomplete() {
             return !option.includes('*');
         });
         $(this).autocomplete("option", "source", filteredIndexValues);
-    }).on('keypress', function(event) {
+    }).on('keypress', async function(event) {
         // Clear the input field if the typed value does not match any options when Enter is pressed
         if (event.keyCode === 13) {
             let typedValue = $(this).val();
             if (indexValues.includes(typedValue)) {
                 addSelectedIndex(typedValue);
-                if (!selectedSearchIndex.includes(typedValue)) {
+                if (!selectedSearchIndex.split(',').includes(typedValue)) {
                     selectedSearchIndex += selectedSearchIndex ? ',' + typedValue : typedValue;
                 }
                 $(this).val('');
                 this.style.width = '5px';
-                runQueryBtnHandler();
                 
                 // Remove the selected value from indexValues
                 const index = indexValues.indexOf(typedValue);
                 if (index !== -1) {
                     indexValues.splice(index, 1);
+                    indexValues = indexValues.filter(option => !option.includes('*')); // Remove options including '*'
                     $(this).autocomplete("option", "source", indexValues);
                 }
-                if (typedValue.endsWith('*')) {
-                    const prefix = typedValue.slice(0, -1); // Remove the '*'
-                    let filteredIndexValues = indexValues.filter(function(option) {
-                        return !option.startsWith(prefix);
-                    });
-                    indexValues = filteredIndexValues;
-                    $(this).autocomplete("option", "source", filteredIndexValues);
-                }
             } else {
+                var matcher = new RegExp( typedValue.replace("*",".*") ,"gi" );
+                function hasMatchingString(arr, regex) {
+                    return arr.some((element) => regex.test(element.toLowerCase()));
+                }
+                const matchesIndex =  hasMatchingString(indexValues, matcher);
+                // If the typed value matches any index, add the option to index list
+                if (matchesIndex) {
+                    indexValues.unshift(typedValue);
```

Comment:
We can remove line 172, so the wildcard option will not appear in the options after it is selected.
<img width="289" alt="image" src="https://github.com/siglens/siglens/assets/71771131/1e16c9e8-59c3-4fdf-a8b0-d399b27d7bf3">